client/pool.c

```diff
@@ -569,8 +571,11 @@ static void close_connection(connection_list_element *connection, const char *me
 	if(conn_data->miner) {
 		--conn_data->miner->connections_count;
 		if(conn_data->miner->connections_count == 0) {
-			conn_data->miner->state = MINER_ARCHIVE;
-		}
+			state = conn_data->miner->state = MINER_ARCHIVE;
+			id = conn_data->miner->id;
```

Comment:
Here id is a struct, = is not overload operator. So here it will have the issue of memory.

Reply:
=, the assignment with struct is legit :grin:

Reply:
Okay, good to know this. Normally I will avoid to use = for such cases.